docs/3-accountability-algorithm.md

```diff
@@ -0,0 +1,184 @@
+# Accountability algorithm
+
+## Main abstractions
+
+In order to simplify the discussion we introduce some abstractions to better identify and classify the message logs received and handled by the monitor.
+
+Each **Message** contains the following information:
+
+- **Type** of the message, *type*
+- **Unique identifier** of the sender of the message, *id*
+- **Round** of the message, *r*
+- **Value** of the message, *v*
+
+A **Vote Set** is the set of all Messages that a process sent and received in a specific round *r*.
+
+A **Height Vote Set** is the set of all Vote Set in a specific height *h* or the set of all Messages that a process sent and received in a specific height *h*.
+
+Tendermint messages have lots of other important information that are used by the consensus protocol but we limit our discussion to the necessary data to run the accountability algorithm.       
+
+## Execution steps
+ 
+We show the detailed steps of the execution of the monitor and the accountability algorithm: 
+ 
+1. Upon a fork, monitor sends a request for validators' height vote sets (following a specific communication protocol) for a given height. Monitor starts waiting for incoming message logs.
+
+2. Validators receives the request from the trusted monitor. If they have the requested height vote set, they send it immediately. 
+If they don't have any logs for the requested height, they can simply ignore the request or, alternatively, send a reply to the monitor to notify they do not have any logs for that height.
```

Comment:
They can not ignore the request in synchronous case.

Reply:
I would prefer to avoid this case of not sending a message log.
It will be better for the presentational purposes to say that correct guy sends whatever it has (possibly nothing).

Reply:
I always refer to the async case in this document, I tried to make it more clear. I also removed that part mentioned in the second comment.